### PR TITLE
SDIT-2319 Add paging

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: server/routes/search/searchController.ts [hunk 1]
```diff
@@ -13,48 +21,126 @@ export default class SearchController {
...
       surname: form.lastName,
-      dateOfBirth: form.dob,
-      ageFrom: Number(form.ageRange?.split('-')[0]),
-      ageTo: Number(form.ageRange?.split('-')[1]),
-      gender: form.gender,
+      dateOfBirth: hasDateField ? [form['dob-year'], form['dob-month'], form['dob-day']].join('-') : undefined,
+      ageFrom: Number(form.age?.split('-')[0]) || Number(form.age) || undefined,
+      ageTo: Number(form.age?.split('-')[1]) || undefined,
+
+      gender: undefined,
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:121]: gender/hdc/lifer are still wip

File: server/services/historicalPrisonerService.ts [hunk 2]
```diff
@@ -18,27 +19,33 @@ export default class HistoricalPrisonerService {
...
-      query: querystring.stringify({ ...prisonersByIdentifiersForm }),
+      query: querystring.stringify({ ...prisonersByIdentifiersForm, page }),
     })
   }
 
-  async findPrisonersByAddressTerms(token: string, addressTerms: string): Promise<PagedModelPrisonerSearchDto> {
+  async findPrisonersByAddressTerms(
+    token: string,
+    prisonersByAddress: FindPrisonersByAddress,
+    page: number = undefined,
```
Comment (karenmillermoj) [inline server/services/historicalPrisonerService.ts:44]: wasn't sure what the "norm" was - whether to set undefined or null (same for form fields)

File: server/utils/pagination.test.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
+import { type Pagination, type LegacyPagination, pagination } from './pagination'
+
```
Comment (karenmillermoj) [inline server/utils/pagination.test.ts:2]: Pulled in from incident reporting

File: server/views/partials/searchForm.njk [hunk 4]
```diff
@@ -95,40 +97,105 @@
...
+        value: "address",
         text: "Other",
+        checked: form.searchType === "address",
         conditional: {
         html: otherHtml
       }
       }
     ]
   }) }}
+
```
Comment (karenmillermoj) [inline server/views/partials/searchForm.njk:127]: These are here from when I added previously - left for now but will move once filtering is added

File: server/routes/search/searchController.ts [hunk 5]
```diff
@@ -13,48 +21,126 @@ export default class SearchController {
...
+          SearchController.toPrisonersByAddress(prisonerSearchForm),
+          req.session.searchParams.page,
+        )
+      default:
+        throw new Error(`Unknown search type: ${prisonerSearchForm.searchType}`)
     }
   }
 
+  getPaginationParams(req: Request, page: PageMetaData): LegacyPagination {
+    // const filters = req.session.searchParams.filters ? `&filters=${req.session.searchParams.filters}` : ''
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:96]: wip

File: server/routes/search/searchController.ts [hunk 6]
```diff
@@ -13,48 +21,126 @@ export default class SearchController {
...
+          req.user.token,
+          SearchController.toPrisonersByAddress(prisonerSearchForm),
+          req.session.searchParams.page,
+        )
+      default:
+        throw new Error(`Unknown search type: ${prisonerSearchForm.searchType}`)
     }
   }
 
+  getPaginationParams(req: Request, page: PageMetaData): LegacyPagination {
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:95]: Not sure if this Pagination is the one we want - it looks like it is but odd that it's Legacy

File: server/@types/express/index.d.ts [hunk 7]
```diff
@@ -5,15 +5,19 @@ export declare module 'express-session' {
     searchType: string
     firstName?: string
     lastName?: string
-    dob: string
-    ageRange?: string
+    'dob-day'?: string
```
Comment (petergphillips) [inline server/@types/express/index.d.ts:8]: seems odd field names - would it be better as `dobDay` etc?

Comment (karenmillermoj) [reply server/@types/express/index.d.ts:8]: I didn't realise it was configurable - from govukDateInput - I'll have a look - can prob declare dob on its own as array will look

File: server/services/historicalPrisonerService.ts (hunk 2 above)
Comment (petergphillips) [reply server/services/historicalPrisonerService.ts:44]: don't know either 😁

File: server/views/pages/search.njk [hunk 8]
```diff
@@ -1,25 +1,19 @@
...
-{% from "govuk/components/radios/macro.njk" import govukRadios %}
-{% from "govuk/components/select/macro.njk" import govukSelect %}
-{% from "govuk/components/table/macro.njk" import govukTable %}
-{% from "govuk/components/tag/macro.njk" import govukTag %}
-{% from "moj/components/filter/macro.njk" import mojFilter %}
 
 {% extends "../partials/layout.njk" %}
 
-{% set pageTitle = "Prisoner Search" %}
+{% set pageTitle = "HPA Prisoner Search" %}
```
Comment (petergphillips) [inline server/views/pages/search.njk:16]: probably wouldn't bother with the `HPA` here - I'm not a great fan of the word `Application` in this project anyway

**Review by petergphillips — APPROVED** [review]
LGTM

File: server/views/pages/search.njk (hunk 8 above)
Comment (karenmillermoj) [reply server/views/pages/search.njk:16]: ha ha - was just replicating the current

**Pull request merged**